### ISSUE TITLE
inject less props instead only one object

### ISSUE THREAD

**Issue by pke** [tonyhb/redux-ui#18]
Props can clutter up pretty fast. The redux-ui decorator adds 4 of them. Since the lib is young I propose a breaking change and inject only one additional prop called `ui`. It contains the 2 properties `key` and `state` and the 2 update functions.

What u think?

**Comment by tonyhb** [comment]
Yeah, I think that's a pretty good idea. It'd also be good to clean up context so that we pass down one context object — this is hidden in the decorator right now.

I'll keep this open as we plan a roadmap to 1.0 to think about this more,.
